ruma-identifiers/src/mxc.rs

```diff
@@ -0,0 +1,119 @@
+//! Matrix-spec compliant mxc:// urls.
+
+use std::{
+    convert::TryFrom,
+    fmt::{self, Display},
+    str::FromStr,
+};
+
+use ruma_identifiers_validation::mxc::validate;
+
+use crate::{Error, MediaId, MediaIdBox, ServerName, ServerNameBox};
+
+/// Matrix-spec compliant mxc:// urls.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
Most of these should probably be implemented in terms of `as_str()` like with the other identifier types.